identity/clients/registry.go

```diff
@@ -183,9 +184,25 @@ func (r *Registry) Validate(client *ClientRegistration, clientSecret string, red
 		// and has no configured redirect URIs.
 		redirectURIOK := false
 		for _, urlString := range client.RedirectURIs {
-			if urlString == redirectURIString {
-				redirectURIOK = true
-				break
+			urlStringParsed, _ := url.Parse(urlString)
+			if client.ApplicationType == "native" && urlStringParsed.Host == "localhost" && urlStringParsed.Scheme == "http" {
+				u, err := url.Parse(redirectURIString)
+				if err != nil {
+					break
+				}
+				host, _, err := net.SplitHostPort(u.Host)
```

Comment:
There is no need to do that, since the `url.Hostname()` function does provide the required value directly. So please get rid of the net import and compare `u.Hostname()` with `localhost` down below.